### ISSUE TITLE
Nuget 5.7 incompatible with VS 2019.8

### ISSUE THREAD

**Issue by iskiselev** [microsoft/dotnet-framework-docker#678]
Looks like currently mcr.microsoft.com/dotnet/framework/sdk:4.8 includes msbuild from Visual Studio 2019 Update 8 and NuGet 5.7. They are not compatible with each other - attempt to build project with VS msbuild after restoring with NuGet will result in errors like:
error NETSDK1005: Assets file '...\obj\project.assets.json' doesn't have a target for '...'. Ensure that restore has run and that you have included '...' in the TargetFrameworks for your project.

**Comment by iskiselev** [comment]
NuGet should be update to 5.8 in docker images to solve the problem.

**Comment by iskiselev** [comment]
For reference: https://developercommunity2.visualstudio.com/t/msbuild-uses-net-50-sdk-or-net-core-sdks-for-net-f/1252149?viewtype=solutions

**Comment by mthalman** [comment]
Verified this can repro with the following modifications to https://github.com/microsoft/dotnet-framework-docker/blob/d2b68d441d71df7615eeb8482419c2f975aa462e/samples/dotnetapp/Dockerfile:

Change `dotnet restore` to `nuget restore` here: https://github.com/microsoft/dotnet-framework-docker/blob/d2b68d441d71df7615eeb8482419c2f975aa462e/samples/dotnetapp/Dockerfile#L9

Change `dotnet build` to `dotnet build --no-restore` here: https://github.com/microsoft/dotnet-framework-docker/blob/d2b68d441d71df7615eeb8482419c2f975aa462e/samples/dotnetapp/Dockerfile#L14

It produces the following error:
```
C:\Program Files\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(241,5): error NETSDK1005: Assets file 'C:\app\dotnetapp\obj\project.assets.json' doesn't have a target for 'net48'. Ensure that restore has run and that you have included 'net48' in the TargetFrameworks for your project. [C:\app\dotnetapp\dotnetapp.csproj]
```

**Comment by MichaelSimons** [comment]
There are a couple things we should check when fixing this issue:
1. Make sure there are utests cover this scenario.
1. Review the NuGet 5.8 release notes for breaking or impactful changes.

**Comment by mthalman** [comment]
This applies to more than just an upgrade of NuGet 5.7 to 5.8.  NuGet 5.7 is only installed in Server Core, version 2009.  But the issue described here can occur on all currently supported Server Core versions which have a [variety of NuGet versions installed](https://github.com/microsoft/dotnet-framework-docker/blob/a425cb4ccc1ec00ca26ee1c784c9ca82a1d0dd0e/eng/nuget-info.json).  The reason they have different NuGet versions is because we've taken a conservative approach of changing the NuGet version in order to avoid any breaking changes.  As each new version of Windows is released, we configure it to target the latest available NuGet version.

So in reality, this is more than just upgrading NuGet 5.7 to 5.8.  It's about upgrading all the NuGet versions across all our supported SDK images to NuGet 5.8.  The earliest version that we'd have to consider is 4.9.  That's the version that Server Core 2019 LTSC is configured to use which is the most used version of our images.  I'm still in the process of tracking down whether there are any breaking changes that would exist with such an upgrade.

**Comment by mthalman** [comment]
I've gone through the list of DCRs in the release notes and haven't found any major breaking changes.  I believe that normal NuGet CLI usage would continue to work.  However, if there are dependencies directly on NuGet assemblies, consuming the API, there is the potential for breaking change.  Most notably: https://github.com/NuGet/Home/issues/7294.

I'm thinking that we might want to proceed more cautiously before making this change for the December update.  I'd like to discuss things first with the NuGet team and figure out an upgrade strategy with them.  Then we can create a proper announcement ahead of time with our plans.

In the mean time, the workaround would be to ensure that `dotnet restore` is used instead of `nuget restore`.

**Comment by iskiselev** [comment]
It is not a problem for me right now, but using `dotnet restore` in place of `nuget restore` will not work for projects that still use packages.config: dotnet/sdk#7922. It makes restoring solution that contains both old and new project very tricky in sdk image until nuget will be updated.

**Comment by mthalman** [comment]
If you make use of packages.config, you can use `msbuild -t:restore -p:RestorePackagesConfig=true`.  That [option](https://docs.microsoft.com/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-with-msbuild) is available in MSBuild 16.5+.

**Comment by mthalman** [comment]
The latest set of images published as part of yesterday's Patch Tuesday release (https://github.com/microsoft/dotnet-framework-docker/pull/714) have included an additional version of nuget.exe (version 5.8) that is available from the following path: `%ProgramFiles%\NuGet\latest\nuget.exe`.  For now, NuGet 5.7 is still available from `%ProgramFiles%\NuGet\nuget.exe`.  This will be updated to 5.8 in the February release.  See the following policy announcement for details on this change: https://github.com/microsoft/dotnet-framework-docker/issues/704.
